xapian-applications/omega/index_file.cc

```diff
@@ -238,6 +238,10 @@ index_add_default_filters()
 		  Filter("numbers2text", SEEK_DEV_STDIN));
     index_command("application/vnd.apply.pages",
 		  Filter("pages2text", SEEK_DEV_STDIN));
+
+    index_command("application/epub+zip",
+      Filter("unzip -p %f '*/*.*htm*'", "text/html", "iso-8859-1",SEEK_DEV_STDIN));
+
 }
```

Comment:
Some style issues here:

* The wrapped argument should be aligned under the first argument, like the lines above (so `F` under the first `"`)
* Missing space after the last `,`
* There's an extra blank line before the closing `}` of the function

As an almost universal rule when modifying existing code, you want to make your change code match the code around it.

Reply:
I see. I downloaded few more including the debianpolicy epub which you used for demonstration. Their mime-types are like : 

`john.epub:   application/epub+zip
kadre.epub:  application/epub+zip
matt.epub:   application/epub+zip
policy.epub: application/zip`

I guess, this means one more mime-type to identify. However I am still thinking how much variation can standard epub files have. Also the application/zip have a lot of xhtml files inside. 

This suggests to me that I have to generalize the epub format. I will add the code for application/zip type, but also would like to ask you what other variations can be observed.

Reply:
I will take care of coding format.

Reply:
This will find the upper as well as the sub-level html files. I also added one more mime-type for epub : appilcation/zip.

Reply:
`appilcation/zip` is a ZIP archive. We definitely **don't** want to treat all files identified as `application/zip` as EPUB.

 EPUB uses the ZIP format as a container (which quite a lot of modern file formats do).  What's presumably happening here is libmagic's test for EPUB is a bit too strict, which results in it failing to spot some EPUB files and so it gives the more generic `application/zip` instead of `application/epub+zip`.

The best way to address this is to simply tell omindex to map files with the extension `.epub` to mime-type `application/epub+zip`, and then libmagic won't get used on such files (which will also be also faster).  The built-in extension to mime-type mappings are in `mimemap.tokens`.

Reply:
I did added the epub to the /etc/mimemap.tokens. I guess it has to be added to the xapian version in xapian-applications/omega/mimemap.tokens.

Reply:
Do you mean `/etc/mime.types`?  The `mimemap.tokens` file is a source file for xapian-omega and doesn't get installed - instead code is generated from it.

We don't look at `/etc/mime.types` - in some cases it'll likely contain mappings for files where we really want to look at the file to determine its true type (e.g. some `.doc` files are actually RTF, so we deliberately don't have a mapping for `.doc` so we'll use libmagic to check what such a file actually is).